src/py/mat3ra/made/tools/analyze/interface.py

```diff
@@ -76,3 +104,114 @@ def substrate_strained_configuration(self) -> SlabStrainedSupercellConfiguration
+    def get_pymatgen_coherent_interface_builder(self) -> CoherentInterfaceBuilder:
+        return CoherentInterfaceBuilder(
+            substrate_structure=to_pymatgen(self.substrate_slab_configuration.atomic_layers.crystal),
+            film_structure=to_pymatgen(self.film_slab_configuration.atomic_layers.crystal),
+            substrate_miller=self.substrate_slab_configuration.atomic_layers.miller_indices,
+            film_miller=self.film_slab_configuration.atomic_layers.miller_indices,
+            zslgen=self.get_pymatgen_zsl_generator,
+        )
+
+    @cached_property
+    def zsl_match_holders(self) -> List[ZSLMatchHolder]:
+        """Get ZSL matches between substrate and film slabs."""
+        zsl_matches = self.get_pymatgen_coherent_interface_builder.zsl_matches
+        match_holders = []
+        for match in zsl_matches:
+            match_holder = ZSLMatchHolder(
+                film_transformation=match.film_transformation,
+                substrate_transformation=match.substrate_transformation,
+            )
+            match_holders.append(match_holder)
+        return match_holders
+
+    def _create_strained_configs(
+        self, match: ZSLMatchHolder
+    ) -> Tuple[SlabStrainedSupercellConfiguration, SlabStrainedSupercellConfiguration]:
+        # Use ZSL transformation matrices as supercell matrices
+        substrate_supercell_matrix = match.substrate_transformation.astype(int)
+        film_supercell_matrix = match.film_transformation.astype(int)
+
+        # Substrate remains unstrained (identity strain)
+        substrate_strain_2d = np.eye(2)
+
+        # Calculate film strain using our existing approach (from InterfaceAnalyzer)
+        film_strain_2d = self._calculate_film_strain_for_supercells(substrate_supercell_matrix, film_supercell_matrix)
+
+        substrate_supercell_2d = SupercellMatrix2DSchema(root=substrate_supercell_matrix.tolist())
+        film_supercell_2d = SupercellMatrix2DSchema(root=film_supercell_matrix.tolist())
+
+        # Convert 2D strain matrices to 3D
+        substrate_strain_3d = np.eye(3)
+        substrate_strain_3d[:2, :2] = substrate_strain_2d
+
+        film_strain_3d = np.eye(3)
+        film_strain_3d[:2, :2] = film_strain_2d
+
+        substrate_config = SlabStrainedSupercellConfiguration(
+            stack_components=self.substrate_slab_configuration.stack_components,
+            direction=self.substrate_slab_configuration.direction,
+            xy_supercell_matrix=substrate_supercell_2d,
+            strain_matrix=Matrix3x3Schema(root=substrate_strain_3d.tolist()),
+        )
+
+        film_config = SlabStrainedSupercellConfiguration(
+            stack_components=self.film_slab_configuration.stack_components,
+            direction=self.film_slab_configuration.direction,
+            xy_supercell_matrix=film_supercell_2d,
+            strain_matrix=Matrix3x3Schema(root=film_strain_3d.tolist()),
+        )
+
+        return substrate_config, film_config
```

Comment:
This function seems too long, we should consider reusing some of the prior functionality for it